lib/smart_answer/calculators/coronavirus_find_support_calculator.rb

```diff
@@ -14,52 +14,72 @@ class CoronavirusFindSupportCalculator
                   :have_you_been_made_unemployed,
                   :mental_health_worries
 
-    def has_results?
-      return false if need_help_with.blank?
-
-      need_help_with != "none"
-    end
+    GROUPS = {
+      feeling_unsafe: [:feel_safe],
+      paying_bills: [:afford_rent_mortgage_bills],
+      getting_food: %i[afford_food get_food],
+      being_unemployed: %i[have_you_been_made_unemployed self_employed],
+      going_in_to_work: %i[worried_about_work are_you_off_work_ill],
+      somewhere_to_live: %i[have_somewhere_to_live have_you_been_evicted],
+      mental_health: [:mental_health_worries],
+    }.freeze
 
-    def needs_help_with?(given_help_item)
-      return false unless has_results?
+    SECTIONS = {
+      feel_safe: lambda { |calculator|
+        calculator.needs_help_with?("feeling_unsafe") && calculator.feel_safe != "yes"
+      },
+      afford_rent_mortgage_bills: lambda { |calculator|
+        calculator.needs_help_with?("paying_bills") && calculator.afford_rent_mortgage_bills != "no"
+      },
+      afford_food: lambda { |calculator|
+        calculator.needs_help_with?("getting_food") && calculator.afford_food != "no"
+      },
+      get_food: lambda { |calculator|
+        calculator.needs_help_with?("getting_food") && calculator.get_food != "yes"
+      },
+      have_you_been_made_unemployed: lambda { |calculator|
+        calculator.needs_help_with?("being_unemployed") && calculator.have_you_been_made_unemployed != "no"
+      },
+      self_employed: lambda { |calculator|
+        calculator.needs_help_with?("being_unemployed") && calculator.self_employed != "yes"
+      },
+      worried_about_work: lambda { |calculator|
+        calculator.needs_help_with?("going_in_to_work") && calculator.worried_about_work != "no"
+      },
+      are_you_off_work_ill: lambda { |calculator|
+        calculator.needs_help_with?("going_in_to_work") && calculator.are_you_off_work_ill == "yes"
+      },
+      have_somewhere_to_live: lambda { |calculator|
+        calculator.needs_help_with?("somewhere_to_live") && calculator.have_somewhere_to_live != "yes"
+      },
+      have_you_been_evicted: lambda { |calculator|
+        calculator.needs_help_with?("somewhere_to_live") && calculator.have_you_been_evicted != "no"
+      },
+      mental_health_worries: lambda { |calculator|
+        calculator.needs_help_with?("mental_health") && calculator.mental_health_worries != "no"
+      },
+    }.freeze
 
-      need_help_with.split(",").include? given_help_item
+    def show_group?(name)
+      GROUPS[name].map { |section| show_section?(section) }.any?
     end
 
-    def needs_help_in?(given_nation)
-      nation == given_nation
+    def show_section?(name)
+      SECTIONS[name].call(self)
```

Comment:
If `SECTION` is removed and each action is defined as a method (as currently) this can be simplified to:
```ruby
def show_section?(name)
  __send__(name)
end
```